test/e2e/es/podtemplate_test.go

```diff
@@ -7,20 +7,110 @@ package es
 import (
 	"bytes"
 	"context"
+	"fmt"
 	"io/ioutil"
 	"net/http"
 	"testing"
 
+	"github.com/blang/semver"
 	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
 	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test"
 	"github.com/elastic/cloud-on-k8s/test/e2e/test/elasticsearch"
 	"github.com/magiconair/properties/assert"
 	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// TestPodTemplateValidation simulates a cluster update with a temporary invalid pod template.
+// The invalid pod template should not prevent the update to be eventually applied.
+func TestPodTemplateValidation(t *testing.T) {
+
+	kubernetesVersion, err := semver.ParseTolerant(test.Ctx().KubernetesVersion)
+	if err != nil {
+		t.Fatalf(
+			"Failed to parse the Kubernetes version: %v, err is %v", test.Ctx().KubernetesVersion, err)
+	}
+	if kubernetesVersion.LT(semver.MustParse("1.15.0")) {
+		t.SkipNow()
+	}
+
+	b := elasticsearch.NewBuilder("podtemplatevalidation").
+		WithESMasterDataNodes(1, corev1.ResourceRequirements{
+			Requests: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceMemory: resource.MustParse("2Gi"),
+			},
+			Limits: map[corev1.ResourceName]resource.Quantity{
+				corev1.ResourceMemory: resource.MustParse("2Gi"),
+			},
+		})
+
+	stepsFn := func(k *test.K8sClient) test.StepList {
+		return test.StepList{}.
+			WithStep(
+				test.Step{
+					Name: "Update Elasticsearch spec with an invalid podTemplate",
+					Test: func(t *testing.T) {
+						b.WithPodTemplate(elasticsearch.ESPodTemplate(
+							corev1.ResourceRequirements{
+								// We are requesting more memory than the limit, this is invalid and prevents the Pods creation.
+								Requests: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceMemory: resource.MustParse("4Gi"),
+								},
+								Limits: map[corev1.ResourceName]resource.Quantity{
+									corev1.ResourceMemory: resource.MustParse("2Gi"),
+								},
+							}),
+						)
+					},
+				}).
+			WithSteps(
+				// Mutate Elasticsearch with an invalid PodTemplate
+				b.UpgradeTestSteps(k),
+			).WithStep(
+			test.Step{
```

Comment:
indentation feels a bit off here?

Reply:
Agreed but this file is formatted with `go fmt` 🤷‍♂️